src/main/java/com/uber/cadence/serviceclient/WorkflowServiceTChannel.java

```diff
@@ -207,6 +215,45 @@ private static String getEndpoint(String service, String method) {
     return buildThriftRequest(apiName, body, null);
   }
 
+  @Override
+  public CompletableFuture<Boolean> isHealthy() {
```

Comment:
The implementation is doing more than the name. Do we have a better way to describe it? 

Alternatively, I think this should be two separate methods: `isHeathy` and `resetEndpoint`

Reply:
`resetSubchannelPeers()` is already public. The reason why we combined them is to make it a one stop shop

Reply:
Better to call it `resetEndpoint`. The interface is not only for TChannel. gRPC is on the way. 

Combining it doesn't sound great with the name, and is unnecessary effort. I would keep it separate and let the caller to decide how to do with healthcheck return false.

Reply:
grpc doesn't (or shouldn't) have this issue, since the underlying library will re resolve the dns if there is a connection failure for you. tchannel doesn't understand DNS which is why we added this functionality. Assuming Cadence's GRPC integration is similar to Temporals, the resets won't be necessary for tchannel.

Reply:
I see. so looks like that's why you only put in WorkflowServiceTChannel not in the interface. 

I am fine with just expose IsHealthy()